monai/metrics/generalized_dice.py

```diff
@@ -20,109 +20,108 @@
+        ValueError: When the `reduction` is not one of MetricReduction enum.
     """
 
     def __init__(
-        self,
-        include_background: bool = True,
-        reduction: MetricReduction | str = MetricReduction.MEAN_BATCH,
-        weight_type: Weight | str = Weight.SQUARE,
+        self, include_background: bool = True, reduction: str = "mean", weight_type: Weight | str = Weight.SQUARE
     ) -> None:
         super().__init__()
         self.include_background = include_background
-        reduction_options = [
-            "none",
-            "mean_batch",
-            "sum_batch",
-            MetricReduction.NONE,
-            MetricReduction.MEAN_BATCH,
-            MetricReduction.SUM_BATCH,
-        ]
-        self.reduction = reduction
-        if self.reduction not in reduction_options:
-            raise ValueError(f"reduction must be one of {reduction_options}")
+        self.reduction = look_up_option(reduction, MetricReduction)
         self.weight_type = look_up_option(weight_type, Weight)
+        self.sum_over_labels = self.reduction == MetricReduction.SUM or self.reduction == MetricReduction.MEAN
 
-    def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor:  # type: ignore[override]
-        """Computes the Generalized Dice Score and returns a tensor with its per image values.
+    def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        """
+        Computes the Generalized Dice Score and returns a tensor with its per image values.
 
         Args:
-            y_pred (torch.Tensor): binarized segmentation model output. It must be in one-hot format and in the NCHW[D] format,
+            y_pred: Binarized segmentation model output. It must be in one-hot format and in the NCHW[D] format,
                 where N is the batch dimension, C is the channel dimension, and the remaining are the spatial dimensions.
-            y (torch.Tensor): binarized ground-truth. It must be in one-hot format and have the same shape as `y_pred`.
+            y: Binarized ground-truth. It must be in one-hot format and have the same shape as `y_pred`.
+
+        Returns:
+            torch.Tensor: Per batch and per class Generalized Dice Score.
 
         Raises:
-            ValueError: if `y_pred` and `y` have less than 3 dimensions, or `y_pred` and `y` don't have the same shape.
+            ValueError: If `y_pred` and `y` have less than 3 dimensions, or `y_pred` and `y` don't have the same shape.
         """
         return compute_generalized_dice(
-            y_pred=y_pred, y=y, include_background=self.include_background, weight_type=self.weight_type
+            y_pred=y_pred,
+            y=y,
+            include_background=self.include_background,
+            weight_type=self.weight_type,
+            sum_over_labels=self.sum_over_labels,
         )
 
-    def aggregate(self, reduction: MetricReduction | str | None = None) -> torch.Tensor:
```

Comment:
Is there any specific reason remove the reduction here? It seems that the keeping this argument maintains the same logic as other metrics.

Reply:
I've removed the reduction here because the value of `sum_over_labels` needs to be set during the `_compute_tensor` and the tensor will already be reduced over labels at that point. 

So I'm not sure how we can handle a different reduction passed to `aggregate` right now - unless we store an intermediary tensor for the `intersection` and `denominator` in the buffer and make the computation in aggregate. 

Do you have any suggestions here?

Reply:
I agree that the inconsistency might be an issue. We could offer lesser options for reduction as a solution - providing only `mean_batch` and `mean` reductions. 

This would mean you can't compute GeneralizedDice per class but that is probably fine as per the original formulation in the paper which reduces over classes by default.

Reply:
Was originally inspired by this implementation: https://lightning.ai/docs/torchmetrics/stable/segmentation/generalized_dice.html

which does provide values per class but the `per_class` flag is defined during init

Reply:
@KumoLiu Do you think we can follow the `deprecated_arg_default` convention here as well for the reduction on aggregate?

Reply:
Sure, we can do it here. Just ensure we can get the exactly same result with before and let users aware if any behaviour will be changed.

Reply:
Hi @surajpaib, I may misunderstand before, seems we don't need to deprecated the reduction in aggregation. sum_over_classes will only sum over class, then if users also want calculate batch, what should they specify?

Reply:
The `sum_over_classes` is only so we can enable computing gen.dice considering all classes in the numerator and denominator - the default setting for gen dice as in the original paper. 

So if the user wants to reduce `mean_batch` then the `sum_over_classes` will be set to False. But if they want to compute `mean_channel` or overall `mean`, this should be set to True as it will consider all classes in the computation. 

P.S: I realize that I need to add `mean_channel` for setting the `sum_over_classes` flag as well. Will do so asap